app/src/main/java/com/g11x/checklistapp/NavigationActivity.java

```diff
@@ -73,12 +75,19 @@ protected void onCreate(Bundle savedInstanceState) {
     final Activity thisActivity = NavigationActivity.this;
     baseLanguageChangeListener = new AppPreferences.LanguageChangeListener(NavigationActivity.this) {
       @Override
-      public void onChanged(String newValue) {
-        thisActivity.recreate();
+      void onChanged(@NonNull Language newLangauge) {
+        Intent intent = getIntent();
+        finish();
+        startActivity(intent);
```

Comment:
This change is what makes the screen stop flickering. Instead of restarting the same activity and causing a re-render the code gracefully finishing then restarts the activity.